src/services/azureBlobStorageService.ts

```diff
@@ -0,0 +1,141 @@
+import { Aborter, BlockBlobURL, ContainerURL, ServiceURL, StorageURL, uploadFileToBlockBlob } from "@azure/storage-blob";
+import Serverless from "serverless";
+import { BaseService } from "./baseService";
+
+/**
+ * Wrapper for operations on Azure Blob Storage account
+ */
+export class AzureBlobStorageService extends BaseService {
+
+  /**
+   * Account URL for Azure Blob Storage account. Depends on `storageAccountName` being set in baseService
+   */
+  private accountUrl: string;
+
+  public constructor(serverless: Serverless, options: Serverless.Options) {
+    super(serverless, options);
+    this.accountUrl = `https://${this.storageAccountName}.blob.core.windows.net`;
+  }
+
+  /**
+   * Upload a file to Azure Blob Storage
+   * @param path Path of file to upload
+   * @param containerName Name of container in Azure Blob storage for upload
+   * @param blobName Name of blob file created as a result of upload
+   */
+  public async uploadFile(path: string, containerName: string, blobName?: string) {
+    const name = blobName || path.replace(/^.*[\\\/]/, "");
+    uploadFileToBlockBlob(Aborter.none, path, this.getBlockBlobURL(containerName, name));
+  };
+  
+  /**
+   * Delete a blob from Azure Blob Storage
+   * @param containerName Name of container containing blob
+   * @param blobName Blob to delete
+   */
+  public async deleteFile(containerName: string, blobName: string): Promise<void> {
+    const blockBlobUrl = await this.getBlockBlobURL(containerName, blobName)
+    await blockBlobUrl.delete(Aborter.none);
+  }
+
+  /**
+   * Lists files in container
+   * @param ext - Extension of files to filter on when retrieving files
+   * from container
+   */
+  public async listFiles(containerName: string, ext?: string): Promise<string[]> {
```

Comment:
Please validate input arguments using `Guard` to make sure they satisfy the method requirements.